Example/Example/Purchase/PurchaseLogicController.swift

```diff
@@ -208,3 +299,66 @@ final class PurchaseLogicController {
   }
 
 }
+
+extension PurchaseLogicController: PayKitObserver {
+  func stateDidChange(to state: PayKitState) {
+
+    print("Cash app state change:", Mirror(reflecting: state).children.first!.label!)
+
+    switch state {
+    case .notStarted,
+      .creatingCustomerRequest,
+      .updatingCustomerRequest,
+      .redirecting,
+      .polling,
+      .apiError,
+      .integrationError,
+      .networkError,
+      .unexpectedError:
+      return
+    case .readyToAuthorize(let request):
+      enableCashButton()
+      cashRequest = request
+    case .approved(let request, let grants):
+      if
+        PurchaseLogicController.cashData == nil ||
+          request.customerProfile == nil ||
+          grants.isEmpty {
+        return
+      }
+
+      let customerProfile = request.customerProfile!
+      let grant = grants.first!
+      let cashData = PurchaseLogicController.cashData!
+
+      quantities = [:]
+      commandHandler(.updateProducts(productDisplayModels))
+
+      let cashTag = customerProfile.cashtag
+      let customerId = customerProfile.id
+
+      let formatter = CurrencyFormatter(currencyCode: currencyCode)
+      let total = grants.reduce(0) { $0 + ($1.action.money?.amount ?? 0) }
+      let amount = formatter.string(from: Decimal(total) / 100)
```

Comment:
Up to you but you can also use string interpolation. For reference: https://www.hackingwithswift.com/articles/178/super-powered-string-interpolation-in-swift-5-0